### PR TITLE
docs: README rewrite

### PR DESCRIPTION
**Pull request by dvsrepo**

### CODE REVIEW

**Comment by dvsrepo** [comment]
@dcfidalgo please make a round of proof reading, revision, adding things etc. The idea would be to merge it tomorrow

File: README.md [hunk 1]
```diff
@@ -1,84 +1,147 @@
...
+
+- **End-to-end**: Most annotation tools treat data collection as a one-off activity at the beginning of each project. In real-world projects, data collection is a key activity of the iterative ML development process. Once a model goes into production, you want to monitor and analyze its predictions and collect more data to improve your model over time. Rubrix is designed to close this gap, enabling you to **iterate as much you need**.
+
+- **User and Developer Experience**: The key to sustainable NLP solutions is to make it easier for everyone to contribute to projects. *Domain experts* should feel comfortable interpreting and annotating data. *Data scientists* should feel free to experiment and iterate. *Engineers* should feel in control of data pipelines. Rubrix optimizes the experience for these core users to **make your teams more productive**. 
+
+- **Beyond hand-labeling**: Classical hand labeling workflows are costly and inefficient, but having humans-in-the-loop is essential. Easily combine hand-labeling with active learning, bulk-labeling, zero-shot models, and weak-supervision into **novel data annotation workflows**.
```
Comment (dcfidalgo) [inline README.md]:
```suggestion
- **Beyond hand-labeling**: Classical hand labeling workflows are costly and inefficient, but having humans-in-the-loop is essential. Easily combine hand-labeling with active learning, bulk-labeling, zero-shot models, and weak-supervision in **novel data annotation workflows**.
```

File: README.md [hunk 2]
```diff
@@ -1,84 +1,147 @@
...
 
-Rubrix focuses on enabling novel, human in the loop workflows involving data scientists, subject matter experts and ML/data engineers. 
+Rubrix is a **production-ready Python framework for exploring, annotating, and managing data** in NLP projects. 
 
-![](docs/images/rubrix_intro.svg)
+Key features:
+
+- **Open**: Rubrix is free, open-source, and 100% compatible with major NLP libraries (Hugging Face transformers, spaCy, Stanford Stanza, Flair, etc.). In fact, you can **use and combine your preferred libraries** without implementing any specific interface.
+
+- **End-to-end**: Most annotation tools treat data collection as a one-off activity at the beginning of each project. In real-world projects, data collection is a key activity of the iterative ML development process. Once a model goes into production, you want to monitor and analyze its predictions and collect more data to improve your model over time. Rubrix is designed to close this gap, enabling you to **iterate as much you need**.
```
Comment (dcfidalgo) [inline README.md]:
```suggestion
- **End-to-end**: Most annotation tools treat data collection as a one-off activity at the beginning of each project. In real-world projects, data collection is a key activity of the iterative process of ML model development. Once a model goes into production, you want to monitor and analyze its predictions, and collect more data to improve your model over time. Rubrix is designed to close this gap, enabling you to **iterate as much as you need**.
```

File: README.md [hunk 3]
```diff
@@ -1,84 +1,147 @@
...
+
+## Example
+
+Let's see Rubrix in action with a quick example: *Bootstraping data annotation with a zero-shot classifier* 
+
+**Why**: 
 
-With Rubrix, you can:
+- The availability of pre-trained language models with zero-shot capabilities (i.e., no need to train it for your own use case) means you can, sometimes, accelerate your data annotation tasks by pre-annotating your corpus with a pre-trained model.
+- The same workflow can be applied if there's a pre-trained "supervised" model which fits your categories but you need to fine-tune for your own use case. For example, fine-tuning a sentiment classifier for a highly specific type of messages.
```
Comment (dcfidalgo) [inline README.md]:
```suggestion
- The same workflow can be applied if there is a pre-trained "supervised" model that fits your categories but needs fine-tuning for your own use case. For example, fine-tuning a sentiment classifier for a very specific type of message.
```

File: README.md [hunk 4]
```diff
@@ -1,84 +1,147 @@
...
+- A zero-shot classifier from the 🤗 Hub: `typeform/distilbert-base-uncased-mnli`
+- A dataset containing news
+- A set of target categories: `Business`, `Sports`, etc.
 
-- a **Python library** to bridge data and models, which you can install via `pip`.
-- a **web application** to explore and label data, which you can launch using Docker or directly with Python.
+**What are we going to do**:
 
+1. Make predictions and log them into a Rubrix dataset.
+2. Use Rubrix UI for filtering records predicted as `Business` with high probability. Use the bulk-labeling feature for labeling 15 records as `Business`.
```
Comment (dcfidalgo) [inline README.md]:
```suggestion
2. Use Rubrix UI for filtering records predicted as `Business` with high probability. Use the bulk-labeling feature to label 15 records as `Business`.
```

File: README.md [hunk 5]
```diff
@@ -1,84 +1,147 @@
...
     )
 
-    rb.log(item, name="ag_news_zeroshot")
+    rb.log(item, name="news_zeroshot")
 ```
 
-# Quick links
+### 2. Explore, Filter and Label
+
+Now we can access our Rubrix dataset and start annotating data. Let's filter the records predicted as `Business` with high probability and use the bulk-labeling feature for labeling 15 records as `Business`:
```
Comment (dcfidalgo) [inline README.md]:
```suggestion
Now we can access our Rubrix dataset and start annotating. Let's filter the records predicted as `Business` with high probability and use the bulk-labeling feature to label 15 records as `Business`:
```

File: README.md [hunk 6]
```diff
@@ -1,84 +1,147 @@
...
+
+https://user-images.githubusercontent.com/1107111/132261244-b9151571-608e-4a41-8f34-e9dc1c8b8e38.mp4
+
+
+### 3. Load and create a training set
+After a round of data annotation, we can load the Rubrix dataset and transform it into a 🤗 dataset for training a supervised classifier.
```
Comment (dcfidalgo) [inline README.md]: Not sure if the transformation to a HF dataset is neccessary, maybe other frameworks prefer DataFrames. I would just explain the steps to filter the annotated records and how to extract the text + label

```suggestion
After a round of data annotation, we can load the Rubrix dataset and create a training set to train a supervised classifier.
```

File: README.md [hunk 7]
```diff
@@ -1,84 +1,147 @@
...
+
+# labels can be a list (for supporting multi-label text classifiers)
+rb_df['labels'] = rb_df.annotation.transform(lambda r: r[0])
+
+
+# create 🤗 dataset from pandas with labels as numeric ids
+label2id = {label: i for i,label in enumerate(labels)}
+train_ds = Dataset.from_pandas(rb_df[['text', 'labels']])
+train_ds = train_ds.map(lambda example: {'labels': label2id[example['labels']]})
+```
```
Comment (dcfidalgo) [inline README.md]: In the master branch, the `annotation` column for `multi_label=False` records, contains strings, not a list of strings. I fixed it in the T2T PR.

```suggestion
```python
# load the Rubrix dataset as a pandas DataFrame
rb_df = rb.load(name='news_zeroshot')

# filter annotated records
rb_df = rb_df[rb_df.status == "Validated"]

# select text input and the annotated label
train_df = pd.DataFrame({
    "text": rb_df.inputs.transform(lambda r: r["text"]),
    "label": rb_df.annotation,
})
```
```

File: README.md [hunk 8]
```diff
@@ -1,84 +1,147 @@
...
+
+## Architecture
+
+Rubrix main components are:
+
+- **Rubrix Python library**: Python client to log, load, copy and delete Rubrix datasets from Python.
+- **Rubrix server**: FastAPI REST service for reading and writing data.
+- **Elasticsearch**: The storage layer and search engine powering the API and UI.
+- **Rubrix UI**: Easy-to-use UI for data exploration and annotation.
```
Comment (dcfidalgo) [inline README.md]: Should we stick with the term `Rubrix UI`? In the docs we often refer to this component as `Web App`. Either way, I think we should be consistent.

**Review by dcfidalgo — COMMENTED** [review]
Nice, big improvement, just a few inline comments

File: README.md (hunk 8 above)
Comment (dvsrepo) [reply README.md]: You're right, can you change it to Web App here?

File: README.md (hunk 6 above)
Comment (dvsrepo) [reply README.md]: Yes, I agree!

**Comment by dvsrepo** [comment]
Thanks a lot @dcfidalgo , not sure if I committed all suggestions, feel free to do so and merge, I'm still on mobile

File: README.md [hunk 9]
```diff
@@ -1,84 +1,143 @@
-
-<p align="center">
-    <img src="docs/images/rubrix_logo.svg" alt="drawing" width="225"/>
+<p align="left">
+    <img src="docs/images/rubrix_logo.svg" alt="drawing" width="200" />
+</p>
+<h4> Python framework to explore, label, and monitor data for NLP</h4>
```
Comment (dvsrepo) [inline README.md]:
```suggestion
<p> Python framework to explore, label, and monitor data for NLP</p>
```

**Pull request merged**